lib/src/app.dart

```diff
@@ -106,6 +112,56 @@ class _AppState extends ConsumerState<Application> {
       setOptimalDisplayMode();
     }
 
+    ref.read(localNotificationServiceProvider).init();
+
+    ref.listenManual(challengesProvider, (prev, current) {
+      final prevIds = prev!.value!.inward.map((challenge) => challenge.id);
+      final inward = current.value!.inward;
+      final repo = ref.read(challengeRepositoryProvider);
+      final l10n = ref.read(l10nProvider).strings;
+
+      inward
+          .map((element) => element.id)
+          .where((id) => !prevIds.contains(id))
+          .map((id) => inward.firstWhere((element) => element.id == id))
```

Comment:
Both `.map()` are unnecessary, you could simply write `.where((element) => !prevIds.contains(element.id))` and remove the maps.